src/modules/ecommerceTracking/buildEcommerceObj.js

```diff
@@ -1,25 +1,28 @@
+import{stringOrNull} from '../stringOrNull';
+
 /*
 param: String, String, String, String, String
 return: Object
 */
 
 //Uses parameters to populate certain properties of buildEcommerceObj when the function is called
-export let buildEcommerceObj = (step, id, affiliation, revenue, products) => {
-    if(typeof step === 'string' && typeof id === 'string' && typeof affiliation === 'string' && typeof revenue === 'string' && Array.isArray(products)) {
+export let buildEcommerceObj = (ecommerceEvent, option, step, currencyCode, id, affiliation, revenue, tax, shipping, products) => {
```

Comment:
This module needs some code refactoring. Although from the technical point of view it does the job I believe the function signature and its conditionals can be rewritten in a better way.

1. Instead of defining all the function parameters, [Rest parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters) could be used.
E.g. `function (x,y,z,etc) to function(...args)`
2. [Destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) `...args`
3. The module `stringOrNull` should accept an Array rather than just a string pass `args` as the parameter.